src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -723,8 +723,12 @@ int xocl_refresh_subdevs(struct xocl_dev *xdev)
 		blob = NULL;
 	}
 
-	if (XOCL_DRM(xdev))
-		xocl_cleanup_mem(XOCL_DRM(xdev));
+	/* clean up mem topology */
```

Comment:
I don't know why we don't call xocl_cleanup_mem, what if the topology has been changed?